concepts/string-vs-str/about.md

```diff
@@ -6,9 +6,66 @@ This document discusses how they overlap and offers some basic advice.
 `String` is an owned, vector of `u8` bytes.
 `String` is mutable.
 `&str`, is a borrowed reference to a String slice.
-`&str` is pronounced "and stir"
 `&str` is immutable.
 
+The proper pronunciation of `&str` is "ref string".
+`String` is a potentially-mutable UTF-8-encoded representation of a sequence of Unicode code points. (In Java or C#, this type is called `StringBuilder`.)
```

Comment:
Do we need this line referencing Java and C#? I'm personally unfamiliar with either of those languages so this does not teach me anything.
It would seem helpful for someone familiar with those languages.
But we generally avoid comparisons in other `about.md` so I'm included to remove this line.

Reply:
Yeah, let's avoid referencing the languages. Turns out [this was already mentioned](https://github.com/exercism/v3/pull/3058/files#r561957164), but got lost at some point among the merges.